suzieq/engines/pandas/network.py

```diff
@@ -179,6 +185,15 @@ def _find_addr_arp(self, addr: str, **kwargs) -> pd.DataFrame:
                      ifname=[oif])
 
             if not ifdf.empty:
+                vrf = ifdf.master.unique().tolist()[0] or 'default'
+                vlan_list = ifdf.vlan.astype(str).unique().tolist()
+                try:
+                    vlan = vlan_list[0]
+                    vlan = int(vlan)
+                except ValueError:
+                    # ignore if the conversion of the vlan to integer has
+                    # failed
+                    pass
```

Comment:
These lines are not clear to me. Why we need to perform cast to string with `.astype(str)` and then we cast back to int in line 192?

Reply:
I actually do not have a response to this. I simply moved the code without modifying it. I guessed it was a problem of NaN values in the unique function, but it seems not.
Maybe @ddutt can give us a better answer

Reply:
This looks like its been added by you @LucaNicosia ?

Reply:
In places where VLAN was not valid such as a routed link, I was putting "-" and so I made the vlan as string. Luca did half a fix :)